Bug report: V crashes after attempting to convert a u8 buffer to a string or a slice with the bytestr() method

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Home v19042 64-bit
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz  

CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.

 cmd: cc --version

getwd: C:\Users\Nath\Documents\vlang\projects\first
vmodules: C:\Users\Nath\.vmodules
vroot: C:\Users\Nath\Documents\vlang\v
vexe: C:\Users\Nath\Documents\vlang\v\v.exe
vexe mtime: 2022-08-26 06:33:00
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 40c0a8c.3af1227

Git version: git version 2.37.1.windows.1
Git vroot status: weekly.2022.34-28-g3af12271
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 125e21e2

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
import os

fn test1(x string, y u32) string {
	return "hey there"
}

fn test2() (u32, u32) {
	return 43, 2
}

pub fn strings() {
	mut out := string("hey there mate")
	println("out.utf8_len: [$out.len_utf8()]\nout.len: [$out.len]\nout.bytes: [$out.bytes()]")
	println("out: [$out]")
	
	slice := out[0..out.len]
	println("out_slice: [$slice]")

	out_clone := out.clone()
	println("clone of out: [$out_clone]")
	println("out.str: [$out.str]")

	mut wide_str := string("some wide string")
	wide_str.to_wide()
	println("wide_str: [$wide_str]")

	mut ascii_str := "some wide string"
	// ascii_str.is_ascii()
	ascii_str.bytes()
	println("ascii_str: [$ascii_str] - ansi_string?: [$ascii_str.is_ascii()]")
	mut vstring := "ansi_string"
	println("vstring is ansi_string?: [$vstring.is_ascii()]")

	wide := 'wide'.runes()
	for w in wide {
		print("$w")
	}

	println("")

	mut arr := ["hey"]
	arr << "there"
	arr << "mate"
	println("$arr")

	for i in arr { println("$i") }
	test_print_array := ["hey", "there", "mate", "whats", "your", "name?"]

	print("can we find the key word 'whats' in dis array? [")
	print("whats" in test_print_array)
	print("]\n")

	mut buffer := []u8{cap: 1024}

        ## The bug is found below
        ##================
        buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()
        print(buffer.bytestr())
	println(buffer)
        ##================
}

fn main() {
	println('hey there')
	println(os.args)
	println(os.file_name('..\\vls.log'))
	println(os.file_size('..\\vls.log'))
	println(os.hostname())

	s := test1("grapes", 56)
	eprintln(s)
	eprintln(56 + 32 - 32)
	sa, sb := test2()
	println(sa)
	println(sb)

	mut sup := string("sup mate")
	println(sup)

	mut age := u8(01)
	println(age)

	mut counter := 0
	counter += 1
	counter ++
	println(counter)
	counter -= 2
	counter = 100
	println(counter)
	counter ++
	println(counter)

	println("sup [$sup] - age: [$age] - counter: [$counter]\n\t=D")
	sup += " whats your name???"
	println("Concat: [$sup]")

	strings()
}

```

**What did you expect to see?**

I used the bytestr() method just as the compiler suggested I should use.
I expected to see a bytes array be converted to a slice or a string.


**What did you see instead?**
The below error message.
```
==================
C:/Users/Nath/AppData/Local/Temp/v_0/main.16433259765382525942.tmp.c:17492: warning: assignment from incompatible pointer type
C:/Users/Nath/AppData/Local/Temp/v_0/main.16433259765382525942.tmp.c:17622: error: invalid operand types for binary operation
C:/Users/Nath/AppData/Local/Temp/v_0/main.16433259765382525942.tmp.c:17622: error: invalid aggregate type for register load
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

## Update:
Apologies.

I realized the bug occurred when I used the "+=" syntax to concatenate a string to a u8 array instead of the "<<" push syntax as mentioned in the documentation.

I guess the compiler didn't pick this up and crashed as a result.

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -361,7 +361,7 @@ pub fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 					node.pos)
 			}
 		}
-		if left_sym.kind == .array && right_sym.kind == .array {
+		if left_sym.kind == .array && right_sym.kind == .array && node.op == .assign {
 			// `mut arr := [u8(1),2,3]`
 			// `arr = [byte(4),5,6]`
 			left_info := left_sym.info as ast.Array

--- a/vlib/v/checker/tests/array_plus_assign_err.out
+++ b/vlib/v/checker/tests/array_plus_assign_err.out
@@ -1,0 +1,8 @@
+./vlib/v/checker/tests/array_plus_assign_err.vv:1:5: warning: unused variable: `buffer`
+    1 | mut buffer := []u8{cap: 1024}
+      |     ~~~~~~
+    2 | buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()
+./vlib/v/checker/tests/array_plus_assign_err.vv:2:1: error: operator `+=` not defined on left operand type `[]u8`
+    1 | mut buffer := []u8{cap: 1024}
+    2 | buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()
+      | ~~~~~~

--- a/vlib/v/checker/tests/array_plus_assign_err.out
+++ b/vlib/v/checker/tests/array_plus_assign_err.out
@@ -1,8 +1,8 @@
-./vlib/v/checker/tests/array_plus_assign_err.vv:1:5: warning: unused variable: `buffer`
+vlib/v/checker/tests/array_plus_assign_err.vv:1:5: warning: unused variable: `buffer`
     1 | mut buffer := []u8{cap: 1024}
       |     ~~~~~~
     2 | buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()
-./vlib/v/checker/tests/array_plus_assign_err.vv:2:1: error: operator `+=` not defined on left operand type `[]u8`
+vlib/v/checker/tests/array_plus_assign_err.vv:2:1: error: operator `+=` not defined on left operand type `[]u8`
     1 | mut buffer := []u8{cap: 1024}
     2 | buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()
       | ~~~~~~

--- a/vlib/v/checker/tests/array_plus_assign_err.vv
+++ b/vlib/v/checker/tests/array_plus_assign_err.vv
@@ -1,0 +1,2 @@
+mut buffer := []u8{cap: 1024}
+buffer += "ipconfig && sudo apt-get install -y some_amazing_package name + command output".bytes()

